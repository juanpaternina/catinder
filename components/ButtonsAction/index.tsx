import { View, StyleSheet } from "react-native";
import { CircleButton } from "@/components";
import { AntDesign } from "@expo/vector-icons";

type Props = {
  discardHandler?: () => void;
  voteHandler?: () => void;
};

export default function ButtonsAction({ discardHandler, voteHandler }: Props) {
  return (
    <View style={styles.container}>
      <CircleButton onPress={discardHandler}>
        <AntDesign testID="close-icon" name="close" size={26} color="#E16359" />
      </CircleButton>
      <CircleButton onPress={voteHandler}>
        <AntDesign
          testID="heart-icon"
          name="heart"
          size={26.5}
          color="#6BD88E"
        />
      </CircleButton>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    width: 154,
  },
});
