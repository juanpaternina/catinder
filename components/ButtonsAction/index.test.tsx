import { render, screen } from "@testing-library/react-native";
import ButtonsAction from "./index";

jest.mock("@expo/vector-icons", () => ({
  AntDesign: "",
}));

describe("ButtonsAction", () => {
  it("renders correctly", () => {
    render(<ButtonsAction />);
    expect(screen.getByTestId("close-icon")).toBeTruthy();
    expect(screen.getByTestId("heart-icon")).toBeTruthy();
    expect(screen).toMatchSnapshot();
  });
});
