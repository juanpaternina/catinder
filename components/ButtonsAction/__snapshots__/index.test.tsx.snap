// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsAction renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": 154,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 27,
        "boxShadow": "0px 10px 16px 0px #BFBFC04D",
        "height": 54,
        "justifyContent": "center",
        "opacity": 1,
        "width": 54,
      }
    }
  >
    <
      color="#E16359"
      name="close"
      size={26}
      testID="close-icon"
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 27,
        "boxShadow": "0px 10px 16px 0px #BFBFC04D",
        "height": 54,
        "justifyContent": "center",
        "opacity": 1,
        "width": 54,
      }
    }
  >
    <
      color="#6BD88E"
      name="heart"
      size={26.5}
      testID="heart-icon"
    />
  </View>
</View>
`;
